modules/javafx.graphics/src/main/java/com/sun/glass/ui/Application.java

```diff
@@ -744,4 +745,22 @@ public final boolean supportsSystemMenu() {
     public static int getKeyCodeForChar(char c) {
         return application._getKeyCodeForChar(c);
     }
+
+    protected int _isKeyLocked(int keyCode) {
+        // Overridden in subclasses
+        return KeyEvent.KEY_LOCK_UNKNOWN;
+    }
+
+    public final Optional<Boolean> isKeyLocked(int keyCode) {
+        checkEventThread();
+        int lockState = _isKeyLocked(keyCode);
+        switch (lockState) {
+            case KeyEvent.KEY_LOCK_OFF:
+                return Optional.of(false);
+            case KeyEvent.KEY_LOCK_ON:
+                return Optional.of(true);
+            default:
+                return Optional.empty();
+        }
```

Comment:
This could be a switch expression:
```
return switch (lockState) {
    case KeyEvent.KEY_LOCK_OFF -> Optional.of(false);
    case KeyEvent.KEY_LOCK_ON -> Optional.of(true);
    default -> Optional.empty();
}
```
Also, do we use a space after `switch`? I see many instances both with and without in the current source.

Reply:
Do we plan to drop support for JDK11 for OpenJFX 17? If not, we'll need to keep to the classic Switch statement, as Switch expressions were only introduced in JDK12 or 13 as a preview.
That said, since 17 is going to be the next LTS release, maybe the plan is to bump the dependency for OpenJFX then?

Reply:
Totally selfish request: It would be great if OpenJFX 17 would retain compatibilty with JDK 11 -- NetBeans allows including JavaFX and will retain JDK 11 compatibility for the foreseable future. I can understand, that at some point, language features will become too important to retain compatibility, but this looks dubious to me.

Reply:
It was a suggestion, so Kevin can choose not to change it.

Formally, OpenJFX N supports JDK N-1, so for 17 we should be able to use features from 16. It was decided a while back not to force-fail on older JDK versions and let it evolve naturally.

While in this case the change is small and might not be worth it alone, many of the recent language changes are geared towards better readability and reduction of possible mistakes, and we will be hurting ourselves in the long run if we don't pick those up. Every PR is relatively small and doesn't require the new features it can take advantage of, but if we go this way then we will be stuck with JDK 11 until something like Valhalla comes out.

Reply:
> Formally, OpenJFX N supports JDK N-1, so for 17 we should be able to use features from 16. It was decided a while back not to force-fail on older JDK versions and let it evolve naturally.

That's the bit of information I was missing, thanks.

Reply:
Bumping the minimum version of the JDK that is required to run the current version of JavaFX requires prior discussion on the openjfx-dev mailing list and a separate JBS Enhancement, like we did when we bumped the minimum to JDK 11 with [JDK-8209966](https://bugs.openjdk.java.net/browse/JDK-8209966) + [JDK-8210093](https://bugs.openjdk.java.net/browse/JDK-8210093). We wouldn't do it in connection with a bug fix such as this.

Having said that, I was already thinking that JavaFX 17 is a good time to consider bumping the minimum to allow using some of the newer JDK language features, so I'll start that discussion on the list soon.

Reply:
> Also, do we use a space after switch? I see many instances both with and without in the current source.

According to the coding guidelines, there should always be a space after operators such as `if`, `for`, `while`, and `switch` (to distinguish them from method calls). If there are occurrences of `switch(expr)` without the space, they are not following the guidelines.